File: ios/Plugin/JailbreakRootPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(JailbreakRootPlugin)
public class JailbreakRootPlugin: CAPPlugin {
    private let implementation = JailbreakRoot()

    @objc func isJailbrokenOrRooted(_ call: CAPPluginCall) {
        if(UIDevice.current.isSimulator) {
            call.resolve([
                "result": false
            ])
        }
        
        call.resolve([
            "result": UIDevice.current.isJailBroken
        ])
    }
    
}

File: android/src/main/java/de/basecom/plugins/jailbreakroot/JailbreakRootPlugin.java
package de.basecom.plugins.jailbreakroot;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "JailbreakRoot")
public class JailbreakRootPlugin extends Plugin {

    private JailbreakRoot implementation = new JailbreakRoot();

    @PluginMethod
    public void isJailbrokenOrRooted(PluginCall call) {
        boolean isDeviceRooted = RootUtil.isDeviceRooted();

        JSObject ret = new JSObject();
        ret.put("result", implementation.isJailbrokenOrRooted(isDeviceRooted));
        call.resolve(ret);
    }
}
